d2/src/main/java/com/linkedin/d2/balancer/servers/ZooKeeperAnnouncer.java

```diff
@@ -139,7 +166,18 @@ public class ZooKeeperAnnouncer implements D2ServiceDiscoveryEventHelper
   private final AtomicLong _warmupClusterMarkDownStartAtRef = new AtomicLong(Long.MAX_VALUE);
 
   // Field to store the dark warm-up time duration in seconds, defaults to zero
-  private int _warmupDuration;
+  private final int _warmupDuration;
+
+  public enum ActOnWeightBreach {
+    // Ignore and no op.
+    IGNORE,
```

Comment:
Isn't this a little redundant to having `_maxWeight` null? Just a thought

Reply:
not really, as IGNORE will increment the metrics.